File: 5424 - Acesso móvel a sistemas de informação/heranca_v1/heranca_v1/Professor.swift
//
//  Professor.swift
//  heranca_v1
//
//  Created by Gonçalo Feliciano on 26/06/2024.
//

import Foundation


class Professor: Pessoa{
    var listaTurmas:[String]
    
    init(nome: String, idade: UInt8, listaTurmas: [String] = []) {
        self.listaTurmas = listaTurmas
        super.init(nome: nome, idade: idade)
    }
    
    override func getInfo() -> String {
        super.getInfo() + "\nLista turmas: \(self.listaTurmas)"
    }
    
    
}

File: 5089 - Programação – Algoritmos/heranca_java/src/Professor.java
public class Professor extends Pessoa{

    public Professor(String nome, int idade) {
        super(nome, idade);
    }

    public void exibirInfo(){
        System.out.println("exibirInfo professor");
    }

    @Override
    public String toString() {
        return "Professor";
    }
}
